webapp/src/ts/modules/messages/messages-content.component.ts

```diff
@@ -0,0 +1,322 @@
+      ])
+      .then(([contactModel, conversation]) => {
+        if (!contactModel && conversation.length) {
+          const firstTaskWithContact = conversation[0].doc.tasks.find((task) => {
+            const message = task.messages && task.messages[0];
+            return message && message.contact && message.contact._id === id;
+          });
+
+          const firstMessageWithContact = firstTaskWithContact.messages.find(message => message.contact._id === id);
+          contactModel = {
+            doc: {
+              name: '',
+              phone: firstMessageWithContact.to
+            }
+          };
+        }
+
+        if (this.selectedConversation && this.selectedConversation.id !== id) {
+          // ignore response for previous request
+          return;
+        }
+
+        this.globalActions.setLoadingShowContent(false);
+        this.messagesActions.setMessagesError(false);
+        const unread = conversation.filter(message => !message.read);
+        this.firstUnread = _minBy(unread, message => message.doc.reported_date);
+        this.messagesActions.updateSelectedConversation({ contact: contactModel, messages: conversation });
+        this.globalActions.setTitle((contactModel && contactModel.doc && contactModel.doc.name) || id);
+        this.markConversationReadIfNeeded();
+        setTimeout(() => this.scrollToUnread());
+      })
+      .catch((err) => {
+        this.globalActions.setLoadingContent(false);
+        this.messagesActions.setMessagesError(true);
+        console.error('Error fetching contact conversation', err);
+      });
+  }
+
+  private updateConversation(options: any = {}) {
+    const selectedId = this.selectedConversation && this.selectedConversation.id;
+
+    if (!selectedId) {
+      return;
+    }
+
+    if (options.change && options.change.deleted) {
+      return this.messagesActions.removeMessageFromSelectedConversation(options.change.id);
+    }
+
+    const skip = options.skip && this.selectedConversation.messages.length;
+    const limit = !options.skip && this.selectedConversation.messages.length;
+
+    if (skip) {
+      this.loadingMoreContent = true;
+    }
+
+    return this.messageContactService
+      .getConversation(selectedId, skip, limit)
+      .then((conversation) => {
+        this.loadingMoreContent = false;
+
+        const newMessageFromUser = conversation.find(message => (
+          message.doc.sent_by === this.userCtx.name &&
+          !this.selectedConversation.messages.find(existent => existent.id === message.id)
+        ));
+
+        this.messagesActions.updateSelectedConversation({ messages: conversation });
+
+        if (options.skip) {
+          this.allLoaded = conversation.length < this.messageContactService.minLimit;
+          this.firstUnread = null;
+        }
+
+        const first = $('.item-content .body #message-content ul > li').filter(':first');
+        this.markConversationReadIfNeeded();
+
+        let scroll:any = false;
```

Comment:
I think it should be find because this code executes when the $('.message-content-wrapper') exists, on db changes or on scroll event from the same element. It's not on the exact init time of the component .

Reply:
It's happened a couple of times, when I was scrolling to the top to load more messages, that this code scrolled to the top sooner somehow and to load more messages I had to scroll lower and scroll up again. 

When that happened I also had this error in the console:
![image](https://user-images.githubusercontent.com/35681649/96552147-0c08ff80-12bc-11eb-94db-cba0c2fbffcc.png)